<a href="https://colab.research.google.com/github/thehapyone/Thesis_Project/blob/master/thesis_work_data_engineering.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
from google.colab import drive
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/drive


In [0]:
import pandas as pd
import numpy as np

## Data Engineering

In [0]:
# Loading the Thesis data

In [0]:
data_repairs = pd.read_csv("drive/My Drive/Colab Notebooks/Thesis/data/PAC_REPAIRS_TRB_20190625_anonymized.csv", low_memory=True)
data1 = pd.read_csv("drive/My Drive/Colab Notebooks/Thesis/data/LVD_DL20190610_0_anonymized.csv", sep=';', low_memory=False)
data2 = pd.read_csv("drive/My Drive/Colab Notebooks/Thesis/data/LVD_DL20190610_1_anonymized.csv", sep=';', low_memory=False)
data3 = pd.read_csv("drive/My Drive/Colab Notebooks/Thesis/data/LVD_DL20190610_2_anonymized.csv", sep=';', low_memory=False)
data4 = pd.read_csv("drive/My Drive/Colab Notebooks/Thesis/data/LVD_DL20190610_3_anonymized.csv", sep=';', low_memory=False)



In [10]:
data_repairs.head(4)

Unnamed: 0;T_CHASSIS;PART_CODE;Record_Count;RepairDate
0  0;O-717289;VO-23083776;1;2019-01-09 00:00:00.0...    
1  1;O-667412;VO-85020372;1;2018-07-28 00:00:00.0...    
2  2;O-736946;VO-20781146;1;2017-08-24 00:00:00.0...    
3  3;O-884007;VO-20784537;1;2019-03-01 00:00:00.0...

In [11]:
print ("All data size - ", data1.shape, data2.shape, data3.shape, data4.shape)

All data size -  (100000, 447) (100000, 447) (100000, 447) (36677, 447)


In [12]:
# fusing all the data together
raw_data = pd.concat([data1, data2, data3, data4])
raw_data.shape

(336677, 447)

the new data contains a total (336677 by 477)

In [13]:
print (raw_data.head(4))

  BIWEEK_SEND_DATE T_CHASSIS  ...  VFE_0008_HAS_PTO           LAST_RUN.1
0       2018-10-01  O-689666  ...                 1  2019-05-14 10:32:00
1       2019-02-01  N-809250  ...                 1  2019-05-14 10:32:00
2       2017-04-15  N-756592  ...                 0  2019-05-14 10:32:00
3       2018-07-01  O-693195  ...                 0  2019-05-14 10:32:00

[4 rows x 447 columns]


In [39]:
# check for missing values
missing_cols = raw_data.isnull().sum()
remaing_rols_after_na = raw_data.dropna(axis=0)
remaing_rols_after_na.shape
print ('No Samples with Missing values: ', raw_data.shape[0] - remaing_rols_after_na.shape[0])

No Samples with Missing values:  169719


In [56]:
# percentage of missing samples
print (' Percentage of missing samples:{0:8.2f}%'.format(((raw_data.shape[0] - remaing_rols_after_na.shape[0])/raw_data.shape[0])*100))

 Percentage of missing samples:   50.41%


In [35]:
# Find the columns with those missing values
missing_cols = missing_cols[missing_cols > 0]
missing_cols.shape

(365,)

We currently have a total of '2521045' missing values in the data. We will need a way to sort that.
'365' columns out of a total 447 columns have missing values.
A total of 169719 out of 336677 samples have missing values.
 Percentage of missing samples:   50.41%

In [0]:
# save combined data as a single csv file
raw_data.to_csv("drive/My Drive/Colab Notebooks/Thesis/data/combined_data.csv", sep=';', index=False)


In [5]:
# confirming the load the new data
data5 = pd.read_csv("drive/My Drive/Colab Notebooks/Thesis/data/combined_data.csv", sep=';', low_memory=False)
data5.shape

(336677, 447)

In [6]:
data5.isnull().sum().sum()

2521045